### PR TITLE
Restore WunderLINQ support

### PR DESCRIPTION
**Pull request by blackboxembedded**
Hi,

A few years back I submitted some PRs(https://github.com/mapsme/omim/pull/13198 & https://github.com/mapsme/omim/pull/12316) to Maps.me to add WunderLINQ hardware control support. They were accepted but at some point after the ability to launch the "backurl" with ESCAPE broke in the Android and iOS apps.

I've recently discovered this incredible project and wanted to fix the launching of the "backurl" with the ESCAPE key and start suggesting it to our users.

The WunderLINQ enables you to use your motorcycle's handlebar controls to control your smartphone. Essentially the device acts as a Bluetooth keyboard. Our apps are also open source and allow you to launch a variety of Navigation Apps along with many other things.  

Here is a video of the integration in action: https://youtu.be/2QHUiheVF5c

### CODE REVIEW

**Review by rtsisyk — APPROVED** [review]
Yes, `backurl` was broken during refactoring. Thanks for fixing!

File: android/src/com/mapswithme/maps/intent/Factory.java [hunk 1]
```diff
@@ -205,6 +205,15 @@ public boolean run(@NonNull MwmActivity target)
 
       if (!result.isSuccess())
         return false;
+      
+      Uri uri = Uri.parse(getUrl());
```
Comment (biodranik) [inline android/src/com/mapswithme/maps/intent/Factory.java:209]:
```suggestion
      final Uri uri = Uri.parse(getUrl());
```

File: android/src/com/mapswithme/maps/intent/Factory.java [hunk 2]
```diff
@@ -205,6 +205,15 @@ public boolean run(@NonNull MwmActivity target)
...
+      String backUrl = uri.getQueryParameter("backurl");
```
Comment (biodranik) [inline android/src/com/mapswithme/maps/intent/Factory.java:210]:
```suggestion
      final String backUrl = uri.getQueryParameter("backurl");
```

File: iphone/Maps/OMaps.plist [hunk 3]
```diff
@@ -81,6 +81,7 @@
 	<array>
 		<string>ms-outlook</string>
 		<string>googlegmail</string>
+		<string>wunderlinq</string>
```
Comment (biodranik) [inline iphone/Maps/OMaps.plist:84]: Why do you think OM needs this line? Is there a CanOpenURL call for the backurl in some place?

**Review by biodranik — CHANGES_REQUESTED** [review]
1. Please also rename back_url to backurl in android/src/com/mapswithme/maps/MwmActivity.java
2. Did you test your changes? Do they work?
3. Please add a missing DCO signature to your commit or PR message.

File: iphone/Maps/OMaps.plist (hunk 3 above)
Comment (blackboxembedded) [reply iphone/Maps/OMaps.plist:84]: Hi, 

This is required to launch our URL when sent via the backurl argument. CanOpenURL is used here: 

[- (void)goBack {](https://github.com/organicmaps/organicmaps/blob/850fcfdad555723dccdf92d65a69c4995264c3f6/iphone/Maps/Classes/MapViewController.mm#L723)

Comment (biodranik) [reply iphone/Maps/OMaps.plist:84]: That's not a flexible approach. Apps should be able to go back without us adding each one of them. What if the canOpenUrl call will be removed? Will it work?

Comment (blackboxembedded) [reply iphone/Maps/OMaps.plist:84]: I'll try without canOpenUrl and will report back

Comment (blackboxembedded) [reply iphone/Maps/OMaps.plist]: It worked.  I'm going to create a new clean branch.

**Comment by blackboxembedded** [comment]
I've redone my PR based on your feedback: https://github.com/organicmaps/organicmaps/pull/3193

**Comment by biodranik** [comment]
You can edit your commits with git rebase, and then force-push the same branch, without the need to always crate a new PR. In this way the history of all conversations will be preserved.

**Pull request closed without merging**